internal/cache/contentcache.go

```diff
@@ -0,0 +1,88 @@
+package cache
+
+import (
+	"bytes"
+	"io"
+	"sync"
+	"fmt"
+
+	"github.com/dgraph-io/ristretto"
+)
+
+type ContentCache struct {
+	mm    *sync.Mutex
+	m     map[string]*sync.Mutex
+	size  int64
+	cache *ristretto.Cache
+}
+
+type ContentCacheCell struct {
+	id   string
+	Data *bytes.Buffer
+}
+
+func NewContentCache(contentCacheSize int64, metrics bool) (*ContentCache, error) {
+	mm := new(sync.Mutex)
+	m := make(map[string]*sync.Mutex)
+	size := contentCacheSize
+	cache, err := ristretto.NewCache(&ristretto.Config{
+		//NumCounters is 10 times estimated max number of items in cache, as suggested in https://pkg.go.dev/github.com/dgraph-io/ristretto@v0.1.1#Config
+		NumCounters: 1e7,
```

Comment:
It is calculated from number of items, not size of items. I don't think there would be 1000000 files in the cache.